util/imageutil/config.go

```diff
@@ -106,19 +116,102 @@ func Config(ctx context.Context, str string, resolver remotes.Resolver, cache Co
 		children,
 	}
 	if err := images.Dispatch(ctx, images.Handlers(handlers...), nil, desc); err != nil {
-		return "", nil, err
+		return nil, err
 	}
-	config, err := images.Config(ctx, cache, desc, platform)
-	if err != nil {
-		return "", nil, err
+
+	res := &ConfigResult{
+		Digest: desc.Digest,
 	}
 
-	dt, err := content.ReadBlob(ctx, cache, config)
-	if err != nil {
-		return "", nil, err
+	// NOTE: most of this logic is butchered from containerd's images.Manifest,
```

Comment:
The large diff is mostly a result of this change ^

Importing this logic from `images.Manifest` is really not great, however, the API it provides is not usable to provide the raw `[]byte` content of the manifest. I can't find an alternative, but maybe someone more familiar with the containerd API might know?